packages/flame_texturepacker/lib/src/texture_packer_sprite.dart

```diff
@@ -97,42 +150,67 @@ class TexturePackerSprite extends Sprite {
     Anchor anchor = Anchor.topLeft,
     Paint? overridePaint,
   }) {
-    if (!rotate) {
-      return super.render(
-        canvas,
-        position: position,
-        size: size,
-        anchor: anchor,
-        overridePaint: overridePaint,
-      );
-    }
     if (position != null) {
       _tmpRenderPosition.setFrom(position);
     } else {
       _tmpRenderPosition.setZero();
     }
 
-    // If the sprite is rotated on the sprite sheet un-rotate it
-    // and adjust the size
-    final auxAnchor = rotate ? Anchor.bottomLeft : anchor;
+    // Get sprite size
+    _tmpRenderSize.setFrom(size ?? srcSize);
+
+    // Calculate topLeft position
+    _tmpRenderPosition.setValues(
+      _tmpRenderPosition.x - (anchor.x * _tmpRenderSize.x),
+      _tmpRenderPosition.y - (anchor.y * _tmpRenderSize.y),
+    );
+
+    // Calculate multiplier value
+    _tmpRenderScale
+      ..setFrom(_tmpRenderSize)
+      ..divide(originalSize);
+
+    // Calculate image size rendered based on packedSize
+    _tmpRenderImageSize
+      ..setFrom(_packedSize)
+      ..multiply(_tmpRenderScale);
 
-    final tempSize = size ?? srcSize;
-    final tempWidth = rotate ? tempSize.y : tempSize.x;
-    final tempHeight = rotate ? tempSize.x : tempSize.y;
+    // Calculate image rendered offset from topLeft position
+    _tmpRenderOffset
+      ..setFrom(offset)
+      ..multiply(_tmpRenderScale)
+      ..add(_tmpRenderPosition);
 
-    _tmpRenderSize.setValues(tempWidth, tempHeight);
+    if (!rotate) {
+      // Calculate and render for non-rotated image, must call function render
+      // from super class with anchor = Anchor.topLeft, because we already
+      // calculated size and position based on Anchor.topLeft to rendered by
+      // super class function.
+      _tmpRenderSize.setFrom(_tmpRenderImageSize);
+      _tmpRenderPosition.setFrom(_tmpRenderOffset);
+      return super.render(
```

Comment:
I'm curious why this branch doesn't need the decorator

Reply:
Because this branch use non-rotated src image.
Which is at this point the position of the canvas should be like this one.
![image](https://github.com/user-attachments/assets/6951ce14-b80b-4bfc-bd93-2076a8f1fd1e)

due the src image should be like this one below..
So we do not need decorator for this branch to rotate the src image..
![image](https://github.com/user-attachments/assets/2179e1f5-b48e-4743-af26-5c9add84511e)

<hr>

But if we use rotated src image..
First we calculate for this position..
![image](https://github.com/user-attachments/assets/6951ce14-b80b-4bfc-bd93-2076a8f1fd1e)

but our src image will be like this
![image](https://github.com/user-attachments/assets/d7b7aea3-d9b2-47e7-bccc-0faf104205e8)

So we need to calculate and render to be this position
![image](https://github.com/user-attachments/assets/53fe2940-be95-40e5-8786-9c62d57694fc)

at last we use the decorator to rotate the previous rendered image..
and the result should be like this below
![image](https://github.com/user-attachments/assets/6951ce14-b80b-4bfc-bd93-2076a8f1fd1e)

Reply:
makes perfect sense, thanks for the explanation :)